### DISCUSSION TITLE
How do I modify the firmware?

### DISCUSSION THREAD

**Discussion by mjl166** [dl9rdz/rdz_ttgo_sonde#339] [TTGO firmware]
I have tried to use a hex editor to change a frequency on the .bin file directly and that worked well, but I'm not sure how to set up the Wi-Fi connection (such as adding the network and passcode) in order to upload data to sondehub.  I saw somewhere that Arduino IDE can be used to modify the code if some libraries are downloaded to it but am not sure how that would work.  I'm new to this stuff and any help would be appreciated.  Thanks!

**Comment by dl9rdz** [comment]
There are many ways, but modifying the firmware is probably not the best coiche.

Please have a look at https://github.com/dl9rdz/rdz_ttgo_sonde/wiki/Installation (Initial configuration) if you have not done so yet. 

After flashing the bin file, the TTGO should be running a WiFi access point. You can connect to that access point and then access the configuration interface using a web browser. That is the usual way how to configure frequencies or configure/activate SondeHub upload.

**Reply by mjl166** [reply]
Thanks so much for your quick reply.  All good now!  It is much more user-friendly than I anticipated.

**Comment by wrevenaugh** [comment]
@dl9rdz - what an awesome piece of software you have created!  Thank you so much for sharing this with the world.  I dabble a little in arduinos and have some coding experience but I'll be darned if I can figure out where to look for your Vertical and Horizontal speed output.  I want to convert them to imperial units (I'm in the US).  I'm looking at the .ino and all the functions in the src directory.  Can you please share with me the best spot in the code for changing the VS and HS to imperial?

**Reply by smithse** [reply]
I could be wrong..C is not my strongest language LOL, but it looks like the sonde type may come into play here??

If you're only wanting to see it on the display, then the void Display::drawVS and void Display::drawHS methods could be a starting place. 

[https://github.com/dl9rdz/rdz_ttgo_sonde/blob/331fa13330800cf9cfa609ab7ca725564e6f300f/RX_FSK/src/Display.cpp#L1270](url)
